History.md

```diff
@@ -22,6 +23,7 @@
 * Liquid::Template.file_system's read_template_file method is no longer passed the context. (#441) [James Reid-Smith]
 * Remove `liquid_methods` (See https://github.com/Shopify/liquid/pull/568 for replacement)
 * Liquid::Template.register_filter raises when the module overrides registered public methods as private or protected (#705) [Gaurav Chande]
+* Overriding to_str on a drop doesn't do anything anymore, override to_s instead
```

Comment:
I don't think it is common to document unintentional changes in the History.md.  I think we could just document that it is fixed in a follow-up PR for the next version.  Otherwise, LGTM